### PR TITLE
[WIP]EC-359: Refactore chart classes

### PR DESCRIPTION
**Pull request by lucasOsti**
| Question      | Answer
| ------------- | ---
| Tickets       | https://issues.ibexa.co/browse/EC-359
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Tests pass?   | yes
| Doc needed?   | no
| License       | [GPL-2.0](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/LICENSE)

#### Checklist:
- [x] Coding standards (`$ composer fix-cs`)
- [x] Ready for Code Review

### CODE REVIEW

File: src/bundle/Resources/public/js/scripts/core/base.chart.js [hunk 1]
```diff
@@ -2,52 +2,122 @@
...
+            const rangeFromCenter = (outerRadius - innerRadius) / 2 + innerRadius;
+            const xPos = x + Math.cos(centerAngle) * rangeFromCenter;
+            const yPos = y + Math.sin(centerAngle) * rangeFromCenter;
+
+            return { xPos, yPos, startAngle, endAngle, diffAngle };
+        },
+        beforeRender: (chart) => {
+            if (chart.config.options.showDataLabel) {
+                const { datasets } = chart.config.data;
+                const pluginTooltips = [];
```
Comment (GrabowskiM) [inline src/bundle/Resources/public/js/scripts/core/base.chart.js:45]:
```suggestion
                const pluginTooltips = datasets.map((dataset, i) => {
                    return (
                        chart.getDatasetMeta(i).data.map((activeSector, j) => {
                            return new Chart.Tooltip(
                                {
                                    dataValue: chart.data.datasets[i].data[j],
                                    _options: chart.options.tooltips,
                                    _active: activeSector,
                                },
                                chart
                            );
                        })
                    );
                });
```

File: src/bundle/Resources/public/js/scripts/core/base.chart.js [hunk 2]
```diff
@@ -2,52 +2,122 @@
...
+
+                datasets.forEach((dataset, i) => {
```
Comment (GrabowskiM) [inline src/bundle/Resources/public/js/scripts/core/base.chart.js:47]: do not use `i` in such cases

File: src/bundle/Resources/public/js/scripts/core/base.chart.js [hunk 3]
```diff
@@ -2,52 +2,122 @@
...
+                    pluginTooltips.push(
+                        chart.getDatasetMeta(i).data.map((activeSector, j) => {
```
Comment (GrabowskiM) [inline src/bundle/Resources/public/js/scripts/core/base.chart.js:49]: or `j`

File: src/bundle/Resources/public/js/scripts/core/base.chart.js [hunk 4]
```diff
@@ -2,52 +2,122 @@
...
-                checkbox.style.borderColor = checkedColor;
-            } else {
-                checkbox.style.backgroundColor = IBEXA_WHITE;
-                checkbox.style.borderColor = IBEXA_COLOR_BASE_DARK;
-            }
+        setOptions(options) {
+            this.options = {
+                ...defaultOptions,
+                ...options,
+                animation: {
```
Comment (GrabowskiM) [inline src/bundle/Resources/public/js/scripts/core/base.chart.js:112]: why this order? It means you can't overwrite `animation`?

**Review by GrabowskiM — CHANGES_REQUESTED** [review]

**Comment by lucasOsti** [comment]
outdate

**Pull request closed without merging**